### PR TITLE
[Model]Add GCN normalization

### PR DESCRIPTION
**Pull request by dddg617**
## Description

Finish the GCN degree nomalization and multi-layer choice.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [X] Code is well-documented
- [X] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: gammagl/layers/conv/gcn_conv.py [hunk 1]
```diff
@@ -42,12 +57,17 @@ class GCNConv(MessagePassing):
     def __init__(self, 
                 in_channels,
                 out_channels,
+                norm = 'none',
```
Comment (Zhanghyi) [inline gammagl/layers/conv/gcn_conv.py:60]: default both

File: gammagl/models/gcn.py [hunk 2]
```diff
@@ -8,41 +9,48 @@ class GCNModel(tlx.nn.Module):
...
-        self.convs = tlx.nn.ModuleList()
-        self.dropout = tlx.layers.Dropout(drop_rate)
-        self.convs.append(GCNConv(feature_dim, hidden_dim))
-        if num_layers>=3:
-            for i in range(num_layers - 2):
-                self.convs.append(GCNConv(hidden_dim, hidden_dim))
-            self.convs.append(GCNConv(hidden_dim, num_class))
-        else:
-            self.convs.append(GCNConv(hidden_dim, num_class))
+        self.conv = nn.ModuleList([GCNConv(feature_dim, hidden_dim, norm = norm)])
```
Comment (Zhanghyi) [inline gammagl/models/gcn.py:36]: num_layers 应该可以只有一层吧

File: gammagl/models/gcn.py [hunk 3]
```diff
@@ -8,41 +9,48 @@ class GCNModel(tlx.nn.Module):
...
+        for _ in range(1, num_layers - 1):
+            self.conv.append(GCNConv(hidden_dim, hidden_dim, norm = norm))
+        self.conv.append(GCNConv(hidden_dim, num_class, norm = 'none'))
```
Comment (Zhanghyi) [inline gammagl/models/gcn.py:39]: 最后一层是none吗

File: examples/gcn/gcn_trainer.py [hunk 4]
```diff
@@ -128,11 +127,12 @@ def main(args):
     parser.add_argument("--n_epoch", type=int, default=200, help="number of epoch")
     parser.add_argument("--hidden_dim", type=int, default=16, help="dimention of hidden layers")
     parser.add_argument("--drop_rate", type=float, default=0.5, help="drop_rate")
+    parser.add_argument("--num_layers", type=int, default=2, help="number of layers")
+    parser.add_argument("-- norm", type=str, default='both', help="how to apply the normalizer.")
```
Comment (Zhanghyi) [inline examples/gcn/gcn_trainer.py:131]: --norm

File: gammagl/layers/conv/gcn_conv.py [hunk 5]
```diff
@@ -57,10 +77,27 @@ def __init__(self,
...
 
-
     def forward(self, x, edge_index, edge_weight=None, num_nodes=None):
         x = self.linear(x)
-        out = self.propagate(x, edge_index, edge_weight=edge_weight, num_nodes=num_nodes)
+        src, dst = edge_index[0], edge_index[1]
+        if edge_weight is None:
+            edge_weight = tlx.ones(shape=(edge_index.shape[1], 1))
+        weights = edge_weight
+        deg = degree(dst, num_nodes=x.shape[0], dtype = tlx.float32)
```
Comment (Zhanghyi) [inline gammagl/layers/conv/gcn_conv.py:86]: 这里是只计算了一次入度吗 dgl的实现需要计算入度和出度

**Review by Zhanghyi — COMMENTED** [review]
GCN的实验部分是否需要重新跑下

File: gammagl/layers/conv/gcn_conv.py (hunk 5 above)
Comment (dddg617) [reply gammagl/layers/conv/gcn_conv.py:86]: 好的，我改一下

File: gammagl/models/gcn.py (hunk 3 above)
Comment (Zhanghyi) [reply gammagl/models/gcn.py:39]: 这个地方层数要求最少两层吗？ 1层Conv是否可以

**Comment by dddg617** [comment]
> GCN的实验部分是否需要重新跑下

已经运行测试过，torch, tensorflow, paddle都可以运行。
